core/lib/spree/app_configuration.rb

```diff
@@ -257,6 +257,12 @@ class AppConfiguration < Preferences::Configuration
     #   @return [Boolean] Indicates if stock management can be restricted by location
     preference :can_restrict_stock_management, :boolean, default: false
 
+    # Allows restricting what currencies will be available.
+    #
+    # @!attribute [r] available_currencies
+    #   @returns [Array] An array of available currencies from Money::Currency.all
+    preference :available_currencies, :array, default: ::Money::Currency.all
```

Comment:
Only concern is that this should _maybe_ be a proc. We load preferences quite early and customizations might be made to `::Money`. (this is really quite an edge case, so I don't know if it's necessary, but can't hurt)